zenml/steps/trainer/utils.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+import torch
+
+from zenml.utils import path_utils
+
+
+def save_test_results(results, output_path):
+    path_utils.create_dir_if_not_exists(output_path)
+    output_path = os.path.join(output_path, 'test_results')
+    with tf.io.TFRecordWriter(output_path) as writer:
+        for example in to_serialized_examples(results):
+            writer.write(example)
+
+    return results
+
+
+def to_serialized_examples(instance) -> tf.train.Example:
```

Comment:
It looks like ive seen this sort of thing now a 100 times :-D can we consolidate into a utils? I think it is also in splitgen or datagen

Reply:
Yes. Unfortunately, even though the resulting structure is a `tf.train.Example`, the structure of the input is different based on the operation, thus the computation is different. I am not 100% sure if we can come up with a single method, which would cover all of the instances.